src/main/kotlin/com/github/mitchellolsthoorn/testgenie/settings/TestGenieSettingsConfigurable.kt

```diff
@@ -23,27 +24,78 @@ class TestGenieSettingsConfigurable : Configurable {
+        settingsComponent!!.criterionWeakMutation = settingsState.criterionWeakMutation
+        settingsComponent!!.criterionOutput = settingsState.criterionOutput
+        settingsComponent!!.criterionMethod = settingsState.criterionMethod
+        settingsComponent!!.criterionMethodNoException = settingsState.criterionMethodNoException
+        settingsComponent!!.criterionCBranch = settingsState.criterionCBranch
+        settingsComponent!!.minimize = settingsState.minimize
     }
 
     /**
      * Checks if the values of the entries in the settings state are different from the persisted values of these entries.
      */
     override fun isModified(): Boolean {
         val settingsState: TestGenieSettingsState = TestGenieSettingsService.getInstance().state!!
-        var modified: Boolean = settingsComponent!!.globalTimeout != settingsState.globalTimeout
-        modified = modified or (settingsComponent!!.showCoverage != settingsState.showCoverage)
+        var modified: Boolean = settingsComponent!!.showCoverage != settingsState.showCoverage
+        modified = modified or (settingsComponent!!.sandbox != settingsState.sandbox)
+        modified = modified or (settingsComponent!!.assertions != settingsState.assertions)
+        modified = modified or (settingsComponent!!.seed != settingsState.seed)
+        modified = modified or (settingsComponent!!.algorithm != settingsState.algorithm)
+        modified = modified or (settingsComponent!!.configurationId != settingsState.configurationId)
+        modified = modified or (settingsComponent!!.clientOnThread != settingsState.clientOnThread)
+        modified = modified or (settingsComponent!!.junitCheck != settingsState.junitCheck)
+        modified = modified or (settingsComponent!!.criterionLine != settingsState.criterionLine)
+        modified = modified or (settingsComponent!!.criterionBranch != settingsState.criterionBranch)
+        modified = modified or (settingsComponent!!.criterionException != settingsState.criterionException)
+        modified = modified or (settingsComponent!!.criterionWeakMutation != settingsState.criterionWeakMutation)
+        modified = modified or (settingsComponent!!.criterionOutput != settingsState.criterionOutput)
+        modified = modified or (settingsComponent!!.criterionMethod != settingsState.criterionMethod)
+        modified = modified or (settingsComponent!!.criterionMethodNoException != settingsState.criterionMethodNoException)
+        modified = modified or (settingsComponent!!.criterionCBranch != settingsState.criterionCBranch)
+        modified = modified or (settingsComponent!!.minimize != settingsState.minimize)
         return modified
     }
 
     /**
      * Persists the modified state after a user hit Apply button.
      */
     override fun apply() {
+        val seed = settingsComponent!!.seed.toLongOrNull()
+        if (settingsComponent!!.seed != "" && seed == null) {
```

Comment:
Are both checks really necessary?

Reply:
Yes, because if its `null`, then the seed is set to `random`

Reply:
But what is the difference between checking for `null` and for empty string?

Reply:
But what is the exact difference between "" and null in this check?